lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -232,6 +233,45 @@ def test_course_about_in_cart(self):
         self.assertEqual(response.status_code, 200)
         self.assertIn(in_cart_span, response.content)
 
+    def assert_enrollment_link_present(self, is_anonymous, _id=False):
+        """ Prepare ecommerce checkout data and assert if the ecommerce link is contained in the response. """
+        checkout_page = '/test_basket/'
+        sku = 'TEST123'
+        CommerceConfiguration.objects.create(
+            checkout_on_ecommerce_service=True,
+            single_course_checkout_page=checkout_page
+        )
+        course = CourseFactory.create(org='new', number='unenrolled', display_name='course')
+        CourseModeFactory(mode_slug=CourseMode.PROFESSIONAL, course_id=course.id, sku=sku, min_price=1)
+
+        request = self.request_factory.get(reverse('about_course', args=[course.id.to_deprecated_string()]))
```

Comment:
In that case I've change all the instances of ``to_deprecated_string()`` in this file.